src/main/java/game/Scene.java

```diff
@@ -1,19 +1,48 @@
 package game;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
-public class Scene implements Interactable {
+public class Scene extends View {
 
     private int id;
     private String name;
     private String description;
     private List<String> items;
-    private Actions actions;
+    private List<String> characters;
+    private Map<String, Integer> nextScenes;
+    private int actionsId;
+    private transient Actions actions;
+
+    public Scene(int id, String name, String description, List<String> items, List<String> characters, Map<String, Integer> nextScenes, int actionsId) {
+        this.id = id;
+        this.name = name;
+        this.description = description;
+        this.items = items;
+        this.characters = characters;
+        this.nextScenes = nextScenes;
+        this.actionsId = actionsId;
+    }
 
     /*
     User Interaction Methods
      */
 
+    @Override
+    void onEnter(Game game, Callback callback) {
+        if (actions == null) {
+            actions = game.getActionsMap().get(actionsId);
+        }
+        callback.onMessage("Entered scene: " + name);
+        callback.onMessage(inspect(new Command(game, "inspect")));
+    }
+
+    @Override
+    void onLeave(Callback callback) {
+        callback.onMessage("You leave " + name);
+    }
+
     /**
      * onCommandEntered is called when the user enters a command and determines if the command can be handled by this
      * Scene.
```

Comment:
Good job in using comments. Makes it easier to understand some parts of the code!